src/pipecat/services/mcp_service.py

```diff
@@ -0,0 +1,213 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+from loguru import logger
+
+from pipecat.adapters.schemas.function_schema import FunctionSchema
+from pipecat.adapters.schemas.tools_schema import ToolsSchema
+
+try:
+    from mcp import ClientSession, StdioServerParameters, types
+    from mcp.client.session import ClientSession
+    from mcp.client.sse import sse_client
+    from mcp.client.stdio import stdio_client
+except ModuleNotFoundError as e:
+    logger.error(f"Exception: {e}")
+    logger.error("In order to use an MCP client, you need to `pip install pipecat-ai[mcp]`.")
+    raise Exception(f"Missing module: {e}")
+
+
+class MCPClient:
+    def __init__(
+        self,
+        server_params: Union[StdioServerParameters, str] = None,
+        pipecat_session: Optional[any] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self._server_params = server_params
+        self._session = ClientSession
+        self._pipecat_session = pipecat_session
+        if isinstance(server_params, StdioServerParameters):
+            self._client = stdio_client
+            self._register_tools = self._stdio_register_tools
+        elif isinstance(server_params, str):
+            self._client = sse_client
+            self._register_tools = self._sse_register_tools
+        else:
+            logger.exception(
+                f"{self} error - server_params must be either StdioServerParameters or an sse server url string."
+            )
```

Comment:
I think this could just raise an exception:

```
raise TypeError(f"{self} invalid argument type: server_params must be either StdioServerParameters or an sse server url string.")
```